GVFS/GVFS.Common/Cleanup/GitCleanupQueue.cs

```diff
@@ -8,70 +8,26 @@ namespace GVFS.Common.Cleanup
 {
     public class GitCleanupQueue
     {
-        private readonly object threadLock = new object();
-        private readonly object currentStepLock = new object();
         private GVFSContext context;
-        private ConcurrentQueue<GitCleanupStep> queue;
-        private Thread thread;
+        private BlockingCollection<GitCleanupStep> queue = new BlockingCollection<GitCleanupStep>();
+        private CancellationTokenSource cancellationToken = new CancellationTokenSource();
         private GitCleanupStep currentStep;
-        private bool stopping;
 
         public GitCleanupQueue(GVFSContext context)
         {
             this.context = context;
-            this.queue = new ConcurrentQueue<GitCleanupStep>();
+            new Thread(() => this.RunQueue()).Start();

```

Comment:
Not something we do much in this code, but something that might be nice, is that we can [name our threads](https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread.name?view=netcore-2.1). This might help when debugging, if we can see the friendly thread name for long running threads.